REST/tests/RESTListenerTest.cc

```diff
@@ -107,14 +108,32 @@ class C4RESTTest
         C4Log("---- %s %s", method.c_str(), uri.c_str());
         string               scheme = config.tlsConfig ? "https" : "http";
         auto                 port   = c4listener_getPort(listener());
-        unique_ptr<Response> r(new Response(scheme, method, requestHostname, port, uri));
-        r->setHeaders(headers).setBody(body);
-        if ( pinnedCert ) r->allowOnlyCert(pinnedCert);
+        unique_ptr<Response> r;
+        unsigned             totalAttempt = 5;
+        double               timeout      = 5;
+        for ( unsigned attemptCount = 0; attemptCount < totalAttempt; ++attemptCount ) {
+            r.reset(new Response(scheme, method, requestHostname, port, uri));
+            r->setHeaders(headers).setBody(body);
+            if ( pinnedCert ) r->allowOnlyCert(pinnedCert);
 #    ifdef COUCHBASE_ENTERPRISE
-        if ( rootCerts ) r->setRootCerts(rootCerts);
-        if ( clientIdentity.cert ) r->setIdentity(clientIdentity.cert, clientIdentity.key);
+            if ( rootCerts ) r->setRootCerts(rootCerts);
+            if ( clientIdentity.cert ) r->setIdentity(clientIdentity.cert, clientIdentity.key);
 #    endif
-        if ( !r->run() ) C4LogToAt(kC4DefaultLog, kC4LogWarning, "Error: %s", c4error_descriptionStr(r->error()));
+            r->setTimeout(timeout);
```

Comment:
Is this request timeout? Not sure if 5 seconds is too low or not. Normaly request timeout is like 60 seconds.

Reply:
This is all on localhost so 5 seconds is probably plenty